install/lsstsw.rst

```diff
@@ -89,7 +89,15 @@ See :doc:`setup` for more information.
 =======================================
 
 Once the LSST Science Pipelines are installed, you can verify that it works by :doc:`running a demo project <demo>`.
-This demo processes a small amount of SDSS data.
+
+5. Upgrading your installation
+==============================
+
+You can upgrade an lsstsw installation in-place by running the following commands from within your :file:`lsstsw/` directory. Given that there may be ABI incompatibilities across conda environments, we suggest removing all of the sub-directories in your :file:`stack/VERSION/OS/` (where ``VERSION`` is the environment version and ``OS`` is the operating system name e.g. ``Linux64``) and `stack/ups_db/` paths.
```

Comment:
I don't think we do require that deletion though. The design of `rebuild` is such that it's meant to allow you to use `envconfig` to switch back to an older conda env and then setup the old packages that used that env.

Reply:
I did say "suggest". Do you have an alternate wording? I know I've run into the occasional ABI problem because I didn't get my env right, and I do this `rm -rf stack/VERSION/Linux64` periodically to clean things up.

Reply:
I don't know. I've never done it so "suggest" seems strong. "if you run into problems you can safely delete blah if you never intend to use the previous builds" is the situation isn't it?  To use older builds it does require that the build number you use for `setup` corresponds to the env that was used to build it and `envconfig` supports that functionality:

```
Usage: /Volumes/MediaHD/Users/timj/work/lsstsw/bin/envconfig  [-i] [-n] [-b] [-h]

Specific options:
-i          interactive, choose from a list of available environments
-n NAME     activate the environment name
-b bXXXX    activate the environment used for the build with id bXXXX
-h          show this message
```

with the `-b` flag.

Reply:
Isn't it sufficient to clean things out by deleting `stack/VERSION/` ?

The ABI compatibilities statement should be removed because the `envconfig` and `rebuild` commands already know that different rubin-envs lead to incompatibilities and so enforce an entire rebuild of the stack when the version changes (that's why version is in the stack path).

Reply:
I believe I've gotten errors when I deleted that whole directory tree. I think something assumes the OS part of the tree already exists by the time you run `rebuild`.

Reply:
This morning when I updated to 0.3.1 I deleted the entire stack/0.1.5 directory and everything is working fine.

Reply:
Ok, thanks. My memory may have been of the pre-condaenv stack.